src/xunit.netcore.extensions/Discoverers/ConditionalTestDiscoverer.cs

```diff
@@ -48,16 +48,36 @@ internal class ConditionalTestDiscoverer
                     continue;
                 }
 
-                string[] symbols = conditionMemberName.Split('.');
                 Type declaringType = testMethodDeclaringType;
 
-                if (symbols.Length == 2)
+                // We have qualified type name with the assembly name, something like
+                // [ConditionalFact("System.PlatformDetection, CoreFx.Private.TestUtilities!" + nameof(PlatformDetection.IsNonZeroLowerBoundArraySupported))]
+                // We don't use '.' as separator in such case and use '!' because the qualified type name can have '.' to include the type namespace.
+                if (conditionMemberName.IndexOf('!') > 0)
                 {
-                    conditionMemberName = symbols[1];
-                    ITypeInfo type = testMethod.TestClass.Class.Assembly.GetTypes(false).Where(t => t.Name.Contains(symbols[0])).FirstOrDefault();
-                    if (type != null)
+                    // get the method name
+                    string[] symbols = conditionMemberName.Split('!');
+                    if (symbols.Length == 2)
```

Comment:
I think if the length is 1 it didn't change the actual behavior, since the default type is the test class (https://github.com/tarekgh/buildtools/blob/663eada3fea68d68e4a1768bf56afaea2e48c54b/src/xunit.netcore.extensions/Discoverers/ConditionalTestDiscoverer.cs#L51) 

Also at the beginning it is assuming that the ConditionalName is the one found as a parameter (https://github.com/tarekgh/buildtools/blob/663eada3fea68d68e4a1768bf56afaea2e48c54b/src/xunit.netcore.extensions/Discoverers/ConditionalTestDiscoverer.cs#L43)

Afterwards it will try to construct the type and method name if it finds a `!` or a `.` and look it up after the calculation: https://github.com/tarekgh/buildtools/blob/663eada3fea68d68e4a1768bf56afaea2e48c54b/src/xunit.netcore.extensions/Discoverers/ConditionalTestDiscoverer.cs#L85

So I don't think this will hurt the existing behavior, but we will gain another way to filter out tests without having to wrap a PlatformDetection property inside a local property.